spec/services/timeline_events/was_last_target_service_spec.rb

```diff
@@ -34,51 +34,77 @@
   end
 
   describe "#was_last_target?" do
-    subject { described_class.new(submission).was_last_target? }
+    subject { described_class.new(submission) }
 
     context "when all assignments are completed by the student" do
       let(:submission) { complete_target(individual_target, student) }
 
       before do
-         student.update(team_id: nil)
-         complete_target(team_target, student)
+        student.update(team_id: nil)
+        complete_target(team_target, student)
       end
 
       it "returns true" do
-        expect(subject).to be true
+        expect(subject.was_last_target?).to be true
       end
 
       it "returns false when a submission is archived" do
         submission.update!(archived_at: Time.zone.now)
-        expect(subject).to be false
+        expect(subject.was_last_target?).to be false
       end
     end
 
     context "when assignments are completed by all team members" do
       let(:submission) { complete_target(team_target, student) }
 
-      before { team.students.each { |s| complete_target(individual_target, s) } }
+      before do
+        team.students.each { |s| complete_target(individual_target, s) }
+      end
 
       it "returns true" do
-        expect(subject).to be true
+        expect(subject.was_last_target?).to be true
       end
     end
 
     context "when assignments are completed only by some team members" do
       let(:submission) { complete_target(team_target, student) }
 
       it "returns false" do
-        expect(subject).to be false
+        expect(subject.was_last_target?).to be false
       end
     end
 
     context "when there are no milestone assignments" do
-      before { course.assignments.milestone.each { |a| a.update!(archived: true) } }
+      before do
+        course.assignments.milestone.each { |a| a.update!(archived: true) }
+      end
 
       let(:submission) { complete_target(team_target, student) }
 
       it "returns false" do
-        expect(subject).to be false
+        expect(subject.was_last_target?).to be false
+      end
+    end
+
+    context "when a target is archived but its milestone assignment is not" do
+      let!(:archived_target) do
+        create :target,
+               :with_shared_assignment,
+               target_group: target_group,
+               visibility: Target::VISIBILITY_ARCHIVED,
+               safe_to_change_visibility: true,
+               given_milestone_number: 3
+      end
+
+      let!(:submission) { complete_target(individual_target, student) }
```

Comment:
This `submission` is called in the setup of `subject`, so there's no need to use `let!` here.

```suggestion
      let(:submission) { complete_target(individual_target, student) }
```
```suggestion
      let!(:submission) { complete_target(individual_target, student) }
```